pkg/distro/packagesets/loader.go

```diff
@@ -34,15 +38,23 @@ func Load(it distro.ImageType, replacements map[string]string) rpmmd.PackageSet
 	archName := arch.Name()
 	distribution := arch.Distro()
 	distroNameVer := distribution.Name()
-	distroName := strings.SplitN(distroNameVer, "-", 2)[0]
```

Comment:
This really seems like it ought to be a function provided by Distro at some point instead of needing to know about the formatting of the Name() result.

Reply:
Yeah, totally agree. The distroidparser cannot be used unfortunately here because of circular imports. A refactor so that distro returns `distro.ID` instead of Name() might be a good first step but then we should try to also remove releaever and OsVersion along the way so it would require a bit of refactor.